apollo-federation-types/src/javascript/mod.rs

```diff
@@ -27,6 +29,12 @@ pub struct SatisfiabilityResult {
     pub hints: Option<Vec<CompositionHint>>,
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+pub struct CompositionResult {
+    pub supergraph: String,
```

Comment:
Can this not also return CompositionErrors and fail to compose?

Reply:
Yeah, I've updated `experimental_composition` to explicitly return hints on success and errors for satisfiability failures